assets/src/edit-story/components/library/libraryUploadDropTarget.js

```diff
@@ -34,16 +34,41 @@ import {
   UploadDropTargetMessageOverlay,
 } from '../uploadDropTarget';
 import { useUploader } from '../../app/uploader';
+import { useMedia } from '../../app/media';
+import {
+  getResourceFromLocalFile,
+  getResourceFromUploadAPI,
+} from '../../components/library/panes/media/mediaUtils';
 
 const MESSAGE_ID = 'edit-story-library-upload-message';
 
 function LibraryUploadDropTarget({ children }) {
   const { uploadFile } = useUploader();
+  const {
```

Comment:
Why is this only here. This behavior should be the same (but enhanced) for the canvas upload, right?

Reply:
Yeah, it definitely will have the canvas target. I just updated the PR title to [WIP] 😄